<?xml version="1.0"?>

<!-- arduino_control.launch
     Gigatron Arduino control launch file

     @author  Syler Wagner  <syler@mit.edu>
     
     @date    2015-09-11    syler   creation
 
     This launch file runs rosserial_python and control_pub_sub.

     Usage Instructions:
     1. In a terminal window, run:
        roslaunch gigatron arduino_control.launch
-->

<launch>

    <arg name="mac" default="false" /> <!-- control from OS X -->
    <arg unless="$(arg mac)" name="port" default="/dev/ttyACM0" />
    <arg if="$(arg mac)" name="port" default="/dev/tty.usbmodem1421" />

<!-- rosserial_python -->
  <node pkg="rosserial_python" type="serial_node.py" name="serial_node" output="screen">
    <param name="port" value="$(arg port)"/>
    <param name="baud" value="115200"/>
  </node>

  <node pkg="gigatron_hardware" type="arduino_drive_controller" name="arduino_drive_controller" output="screen" />

<!-- turtle sim keyboard remote -->
  <!-- <node pkg="gigatron" type="gigaturtle" name="gigaturtle" output="screen" /> -->
  

    
</launch>
  
  
